manifests/install.pp

```diff
@@ -165,7 +165,26 @@
     # Workaround PUP-5802/PUP-5025
     if ($::operatingsystem == 'Fedora') {
       if $pa_collection == 'PC1' or $pa_collection == 'puppet5' {
-        $dist_tag = "fedoraf${::operatingsystemmajrelease}"
+        # There's three cases here due to some mistakes with how we
+        # set-up our distro tags for Fedora platforms:
+        #   * For newer Fedora platforms (e.g. Fedora 28), we want
+        #     to use the fc<major> tag
+        #
+        #   * For older Fedora platforms (e.g. Fedora 26 and 27), we
+        #     have two separate cases:
+        #       * If the package version's > 5.5.3, then we use the fedora<major>
+        #         tag, b/c in those versions we removed the 'f' prefix.
+        #
+        #       * If the package version's <= 5.5.3, then we use the fedoraf<major>
+        #         tag b/c the 'f' prefix is still there.
+        #     
+        if (versioncmp("${::operatingsystemmajrelease}", '27') > 0) {
+          $dist_tag = "fc${::operatingsystemmajrelease}"
+        } elsif (versioncmp("${package_version}", '5.5.3') > 0) {
+          $dist_tag = "fedora${::operatingsystemmajrelease}"
+        } else {
```

Comment:
Note that the `PC1` collection will always hit the else branch since the agent version's older than `5.5.3`, which is what we want.